cdap-client-tests/src/test/java/co/cask/cdap/client/PreferencesClientTestRun.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.client;
+
+import co.cask.cdap.client.app.FakeApp;
+import co.cask.cdap.client.common.ClientTestBase;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.test.XSlowTests;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Test for {@link PreferencesClient}
+ */
+@Category(XSlowTests.class)
+public class PreferencesClientTestRun extends ClientTestBase {
+
+  private PreferencesClient client;
+  private ApplicationClient appClient;
+
+  @Before
+  public void setUp() throws Throwable {
+    super.setUp();
+    client = new PreferencesClient(clientConfig);
+    appClient = new ApplicationClient(clientConfig);
+  }
+
+  @Test
+  public void testInstancePreferences() throws Exception {
+    Map<String, String> propMap = client.getInstancePreferences();
+    Assert.assertEquals(ImmutableMap.<String, String>of(), propMap);
+    propMap.put("k1", "instance");
+    client.setInstancePreferences(propMap);
+    Assert.assertEquals(propMap, client.getInstancePreferences());
+
+    File jarFile = createAppJarFile(FakeApp.class);
+    appClient.deploy(jarFile);
+
+    propMap.put("k1", "namespace");
+    client.setNamespacePreferences(Constants.DEFAULT_NAMESPACE, propMap);
```

Comment:
Can you add a test to set preferences in one namespace and make sure they are not available in another namespace?